### ISSUE TITLE
Telemetry client custom events not populating client_Browser and operation_Name

### ISSUE THREAD

**Issue by petergphillips** [microsoft/ApplicationInsights-Java#1468]
### Expected behavior
Telemetry client custom events are logged with detailed client information

### Actual behavior
After upgrade to 3.0.2 the following fields are blank:
```
client_Model
client_OS
client_Browser
operation_Name
```

### To Reproduce
Create an event e.g.
```
import com.microsoft.applicationinsights.TelemetryClient
...
telemetryClient.trackEvent("Some event", Map.of("count", String.valueOf(count)), null);
```

Then look in app insights for the events:
```
customEvents
| order by timestamp desc
| project sdkVersion, operation_Name, client_OS, client_Browser, client_Model
```

Under 3.0.2 only the `sdkVersion` is populated, rest of the fields are blank.

#### Sample Application
Happy to provide a sample application which just calls `TelemetryClient.trackEvent` if required.

SDK Version: 2.6.2
Java Agent Version: 3.0.2
OS type and version: Debian 14
Application Server type and version (if applicable): Spring Boot 2.4.2
Using spring-boot? yes

**Comment by trask** [comment]
Hey @petergphillips! I suspect this is the same issue as #1452. Can you try out the SNAPSHOT release link I posted in that issue to see if it addresses your issue?

**Comment by petergphillips** [comment]
Hi @trask, just given it a go.  It is now populating `client_OS` but none of the other fields.
![Screenshot from 2021-02-02 10-02-32](https://user-images.githubusercontent.com/5099053/106584319-f6070e00-653d-11eb-86ac-efadbfa29f1e.png)

**Comment by trask** [comment]
hey @petergphillips, how are you setting those other fields? e.g. directly during construction of the telemetry

```
RequestTelemetry requestTelemetry = new RequestTelemetry(...);
requestTelemetry.getContext().getOperation().setName("my-operation-name");
telemetryClient.trackRequest(requestTelemetry);
```

or via a telemetry processors/initializer?

**Comment by petergphillips** [comment]
neither, they are set automatically for us by the application insights agent.  Note that the fields all appear correctly under all 3 versions in `requests`, but only 2.6.2 then populates `customEvents` correctly too.  IIRC there was a setting in application insights that allows the client information to be displayed, as by default application insights doesn't show some of those fields as it considers them sensitive.  I can create a simple test application demonstrating the issue if required.

**Comment by trask** [comment]
can you attach your ApplicationInsights.xml file (scrubbing your instrumentation key)?

**Comment by petergphillips** [comment]
We aren't using an xml file. Our applicationinsights.json is
```
{
  "role": {
    "name": "our-server"
  },
  "customDimensions": {
    "service.version": "${BUILD_NUMBER}"
  },
  "instrumentation": {
    "logging": {
      "level": "DEBUG"
    }
  },
  "selfDiagnostics": {
    "destination": "console"
  }
}
```

**Comment by petergphillips** [comment]
Our instrumentation key is passed in by setting the `APPLICATIONINSIGHTS_CONNECTION_STRING` environment property

**Comment by trask** [comment]
Thanks, this picture mostly makes sense to me, for telemetry besides `request` telemetry:

![Screenshot from 2021-02-02 10-02-32](https://user-images.githubusercontent.com/5099053/106584319-f6070e00-653d-11eb-86ac-efadbfa29f1e.png)

* `operation_Name` not being populated is a known issue https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-upgrade-from-2x#operation-name-on-dependencies
  * Though I just figured out how to populate it in the 3.0 world, and is part of pending PR #1467 
* `client_*` are (mostly intentionally) only populated on `request` telemetry in 3.0. Do you have a use for them on other telemetry, or were you just noticing the differences?
  * I have no idea how `client_OS` is being reported in `3.0.3-BETA-SNAPSHOT`. I couldn't reproduce that, can you confirm that finding?

**Comment by petergphillips** [comment]
TBH the only one out of the four we are interested in is the `client_Browser`.  I only raised the other three as I started comparing the old and the new to see the differences.

Regarding `client_OS`, the screenshot is from application insights so that's what it is reporting.

Our support was using `client_Browser` to then find out why users weren't able to login,e.g. if the `customEvent` is a login event and the browser is IE11 then they can go back to the user asking them to try Firefox / Chrome instead.  We can write a query for them which joins with `requests` on `operation_Id`  to get the same information.  If it is deliberate then that's fine.
